apps/typegpu-docs/src/content/docs/fundamentals/resolve.mdx

```diff
@@ -0,0 +1,124 @@
+---
+title: Resolve
+description: Resolve API can be used to extend shader code with WGSL definitions of TypeGPU-defined objects.
+---
+
+Defining shader schemas and objects in JS/TS has lots of benefits, but for a while it meant introducing significant duplication between host and GPU code.
+In addition to that it was crucial to keep both sides in sync manually when making any changes.
```

Comment:
Anybody can hop into this guide at random, so the concept of time ("for a while") can be confusing.
If it was meant to refer to prior versions of TypeGPU, imo these guides should be "timeless", meaning as if these features always existed.

```suggestion
Defining shader schemas and objects in JS/TS has lots of benefits, but having to keep them in sync with the corresponding WGSL code is hard to maintain.
```